numpy/core/fromnumeric.py

```diff
@@ -1795,6 +1795,12 @@ def nonzero(a):
         Equivalent ndarray method.
     count_nonzero :
         Counts the number of non-zero elements in the input array.
+        
+    Notes
+    -----
+    To obtain the non-zero values, it is recommended to use ``x[y.astype(bool)]`` 
+    which will correctly handle 0-d arrays. In most cases y is already a boolean 
+    array, so astype is not necessary. 
```

Comment:
Shouldn't this be `x[x.astype(bool)]` without the `y`?

Reply:
Done

Reply:
Now the `y` in the second line is wrong. Also since we now are using `x`, the sentence beginning "In most cases ..." since we don't know anything about `x`. Probably should drop that.

Reply:
Done. It seems like we came back from where we started.